libcalico-go/lib/backend/k8s/resources/node.go

```diff
@@ -230,7 +230,20 @@ func K8sNodeToCalico(k8sNode *kapiv1.Node, usePodCIDR bool) (*model.KVPair, erro
 	annotations := k8sNode.ObjectMeta.Annotations
 	bgpSpec.IPv4Address = annotations[nodeBgpIpv4AddrAnnotation]
 	bgpSpec.IPv6Address = annotations[nodeBgpIpv6AddrAnnotation]
-	bgpSpec.RouteReflectorClusterID = annotations[nodeBgpCIDAnnotation]
+	// Validate the IP address specified for route reflector.
+	// When route reflector is configured using calicoctl, the address
+	// is already validated. When the node is annotated directly without
+	// using calicoctl, the IP address specified must be validated here.
+	routeReflectorClusterID := annotations[nodeBgpCIDAnnotation]
```

Comment:
We're going to need this logic for all of the annotations eventually. I might do it like this for re-use:

```go
type validatorFunc func(string) error

func getAnnotation(key string, validator validatorFunc) string
```

and then

```go
bgpSpec.RouteReflectorClusterID = getAnnotation(nodeBgpCIDAnnotation, validateIPv4Address)
```